docs/content/dagster-cloud/developing-testing/environment-variables-and-secrets.mdx

```diff
@@ -0,0 +1,284 @@
+- **May be a good fit if...** - A high-level summary of when the method may be a good fit
+
+<table
+  className="table"
+  style={{
+    width: "100%",
+  }}
+>
+  <thead>
+    <tr>
+      <th
+        style={{
+          width: "15%",
+        }}
+      >
+        Setup method
+      </th>
+      <th
+        style={{
+          width: "40%",
+        }}
+      >
+        How it works
+      </th>
+      <th>May be a good fit if...</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>GitHub secrets</td>
+      <td>
+        Environment variables are defined in a GitHub repository using Action
+        Secrets. <strong>Supported for Serverless deployments only</strong>.
+      </td>
+      <td>
+        <ul
+          style={{
+            marginTop: "0px",
+          }}
+        >
+          <li
+            style={{
+              marginTop: "0px",
+            }}
+          >
+            TODO: not sure if we're keeping this method
+          </li>
+        </ul>
+      </td>
+    </tr>
+    <tr>
+      <td>dagster-cloud CLI</td>
+      <td>
+        Environment variables are passed in using the <code>--env</code> flag in
+        the CLI. <strong>Supported for Serverless deployments only</strong>.
+      </td>
+      <td>
+        <ul
+          style={{
+            marginTop: "0px",
+          }}
+        >
+          <li
+            style={{
+              marginTop: "0px",
+            }}
+          >
+            You're using{" "}
+            <a href="/dagster-cloud/deployment/serverless">
+              Serverless deployment
+            </a>
+          </li>
+          <li>You're not using Dagster's GitHub app</li>
+          <li>
+            You want to manage environment variables outside of Dagster Cloud
+          </li>
+        </ul>
+      </td>
+    </tr>
+    <tr>
+      <td>Agent configuration</td>
+      <td>
+        Environment variables are defined in the agent's configuration in your
+        project's <code>dagster_cloud.yaml</code> file. Values are pulled from
+        your user cluster, meaning they don't pass through Dagster Cloud.
+        <strong>Supported for Hybrid deployments only</strong>.
+      </td>
+      <td>
+        <ul
+          style={{
+            marginTop: "0px",
+          }}
+        >
+          <li
+            style={{
+              marginTop: "0px",
+            }}
+          >
+            You're using{" "}
+            <a href="/dagster-cloud/deployment/hybrid">Hybrid deployment</a>
+          </li>
+          <li>
+            You don't want Dagster Cloud to know the values of your environment
+            variables
+          </li>
+        </ul>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+### Reserved variables
+
+[Built-in Dagster Cloud environment variables](#built-in-environment-variables) are reserved and are unavailable for use. If used, the following error will be raised: <TODO></TODO>
```

Comment:
there aren't any reserved variables currently until we land the native secrets UI - so I think we can leave this section out for this PR